kraken-build/src/kraken/std/python/buildsystem/uv.py

```diff
@@ -221,7 +220,9 @@ def get_managed_environment(self) -> ManagedEnvironment:
 
     def update_lockfile(self, settings: PythonSettings, pyproject: TomlFile) -> TaskStatus:
         command = ["uv", "lock", "--package"]
-        sp.check_call(command, cwd=self.project_directory)
+        logger.debug("Run '%s' in %s.", " ".join(command), self.project_directory)
+        env_vars = UvIndexes.from_package_indexes(settings.package_indexes.values()).to_env()
```

Comment:
Could we make this a function /method to be sure we never forget to bundle these together ?

Reply:
Great suggestion! I reformatted the code to use an utility function for when index credentials need to be injected.